cmd/vela-server/schedule.go

```diff
@@ -0,0 +1,382 @@
+	}
+
+	url := strings.TrimSuffix(r.GetClone(), ".git")
+
+	b := new(library.Build)
+	b.SetAuthor(s.GetCreatedBy())
+	b.SetBranch(r.GetBranch())
+	b.SetClone(r.GetClone())
+	b.SetCommit(commit)
+	b.SetDeploy(s.GetName())
+	b.SetEvent(constants.EventSchedule)
+	b.SetMessage(fmt.Sprintf("triggered for %s schedule with %s entry", s.GetName(), s.GetEntry()))
+	b.SetRef(fmt.Sprintf("refs/heads/%s", b.GetBranch()))
+	b.SetRepoID(r.GetID())
+	b.SetSender(s.GetUpdatedBy())
+	b.SetSource(fmt.Sprintf("%s/tree/%s", url, b.GetBranch()))
+	b.SetStatus(constants.StatusPending)
+	b.SetTitle(fmt.Sprintf("%s received from %s", constants.EventSchedule, url))
+
+	// populate the build link if a web address is provided
+	if len(metadata.Vela.WebAddress) > 0 {
+		b.SetLink(fmt.Sprintf("%s/%s/%d", metadata.Vela.WebAddress, r.GetFullName(), b.GetNumber()))
+	}
+
+	var (
+		// variable to store the raw pipeline configuration
+		config []byte
+		// variable to store executable pipeline
+		p *pipeline.Build
+		// variable to store pipeline configuration
+		pipeline *library.Pipeline
+		// variable to control number of times to retry processing pipeline
+		retryLimit = 5
+		// variable to store the pipeline type for the repository
+		pipelineType = r.GetPipelineType()
+	)
+
+	// implement a loop to process asynchronous operations with a retry limit
+	//
+	// Some operations taken during this workflow can lead to race conditions failing to successfully process
+	// the request. This logic ensures we attempt our best efforts to handle these cases gracefully.
+	for i := 0; i < retryLimit; i++ {
+		logrus.Debugf("compilation loop - attempt %d", i+1)
+		// check if we're on the first iteration of the loop
+		if i > 0 {
+			// incrementally sleep in between retries
+			time.Sleep(time.Duration(i) * time.Second)
+		}
+
+		// send API call to attempt to capture the pipeline
+		pipeline, err = database.GetPipelineForRepo(b.GetCommit(), r)
+		if err != nil { // assume the pipeline doesn't exist in the database yet
+			// send API call to capture the pipeline configuration file
+			config, err = scm.ConfigBackoff(u, r, b.GetCommit())
+			if err != nil {
+				return fmt.Errorf("unable to get pipeline config for %s/%s: %w", r.GetFullName(), b.GetCommit(), err)
+			}
+		} else {
+			config = pipeline.GetData()
+		}
+
+		// send API call to capture repo for the counter (grabbing repo again to ensure counter is correct)
+		r, err = database.GetRepoForOrg(r.GetOrg(), r.GetName())
+		if err != nil {
+			err = fmt.Errorf("unable to get repo %s: %w", r.GetFullName(), err)
+
+			// check if the retry limit has been exceeded
+			if i < retryLimit-1 {
+				logrus.WithError(err).Warningf("retrying #%d", i+1)
+
+				// continue to the next iteration of the loop
+				continue
+			}
+
+			return err
+		}
+
+		// update repo fields with any changes from SCM process
+		r.SetTopics(r.GetTopics())
+		r.SetBranch(r.GetBranch())
```

Comment:
These lines are no longer necessary since we aren't getting any extra info from the SCM for this

Reply:
fixed 😃